### PR TITLE
Overwrite order URL for admins in order details emails

### PR DESCRIPTION
**Pull request by emdashcodes**
See https://github.com/Automattic/wp-calypso/issues/16660.

The emails being sent to the store admin (which is also our only notification right now) links to `wp-admin` instead of Calypso. This PR adds a hotfix for overwriting the templates with our own, linking to WP.com.

To Test:
* Upload/install this branch over your existing `wc-api-dev` plugin.
* Go through the check-out process and make sure you receive an admin order notification email.
* Make sure that the link on the Order # goes to WP.com (make sure you are looking at the admin notification, and not the notification that is sent to the customer).
* To test the plaintext email (which is not currently an option we offer on .com, but I provided here anyway), you can go to `wp-admin > WooCommerce > Settings > Emails > New Order` and then set it to plaintext.
* Repeat the above steps.

### CODE REVIEW

File: hotfixes/wc-api-dev-email-order-url.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @since 0.8.2
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+function wc_api_dev_email_get_wpcom_order_link( $order_id ) {
+	$site_slug = preg_replace( '/^https?:\/\//', '', get_site_url() );
```
Comment (ryelle) [inline hotfixes/wc-api-dev-email-order-url.php]: Probably not something to worry about immediately, but this won't work for URLs where WP is in a subfolder. You can steal Jetpack's [`build_raw_urls` function](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L6371) though (and it looks like this is usually passed `get_home_url`, not `get_site_url`).

**Review by ryelle — COMMENTED** [review]
One comment about the URL builder, but otherwise it seemed to test OK.

My api-dev test site is just pressable, not AT, so I got redirected back to /stats/ when I clicked the email links, but if I switched them to wpcalypso they're fine.

File: hotfixes/wc-api-dev-email-order-url.php (hunk 1 above)
Comment (emdashcodes) [reply hotfixes/wc-api-dev-email-order-url.php]: Good catch and suggestion. I think it's a good idea to handle this now so we don't run into issues later.

**Comment by emdashcodes** [comment]
I'll re-open this after the plugin split.

**Comment by emdashcodes** [comment]
@ryelle I made the change that should allow for sites w/ paths later on. Can you take a look/test and ✅  if everything looks good?

**Review by ryelle — APPROVED** [review]
Code looks good 👍  I don't actually have a subfolder site to test on, but I know that pattern works for jetpack, so I think you're good to merge

**Pull request merged**